ui/app/components/job-version.js

```diff
@@ -108,6 +115,45 @@ export default class JobVersion extends Component {
   })
   revertTo;
 
+  @action async cloneAsNewVersion() {
+    try {
+      this.router.transitionTo(
+        'jobs.job.definition',
+        this.version.get('job.idWithNamespace'),
+        {
+          queryParams: {
+            isEditing: true,
+            version: this.version.number,
+          },
+        }
+      );
+    } catch (e) {
+      this.args.handleError({
+        level: 'danger',
+        title: 'Could Not Edit from Version',
+      });
+    }
+  }
+
+  @action async cloneAsNewJob() {
+    try {
+      let job = await this.version.get('job');
+      let specification = await job.fetchRawSpecification(this.version.number);
+      let specificationSourceString = specification.Source;
+      this.router.transitionTo('jobs.run', {
+        queryParams: {
+          sourceString: specificationSourceString,
+        },
+      });
+    } catch (e) {
+      this.args.handleError({
+        level: 'danger',
+        title: 'Could Not Clone as New Job',
+        description: messageForError(e),
+      });
```

Comment:
To check: how does this handle for specification-doesnt-exist-but-definition-does?